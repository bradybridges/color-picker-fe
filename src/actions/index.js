export const addProjects = (projects) => ({
  type: 'ADD_PROJECTS',
  projects,
});

export const addPalettes = (palettes) => ({
  type: 'ADD_PALETTES',
  palettes,
});
