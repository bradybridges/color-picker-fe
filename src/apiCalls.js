export const getProjects = async () => {
  const response = await fetch('https://color-picker-backend.herokuapp.com/api/v1/projects');
  if(!response.ok) {
    return Error('Failed to fetch projects');
  }
  const projects = await response.json();
  return projects;
};

export const getPalettes = async () => {
  const response = await fetch('https://color-picker-backend.herokuapp.com/api/v1/palettes');
  if(!response.ok) {
    return Error('Failed to fetch palettes');
  }
  const palettes = await response.json();
  return palettes;
};

export const getProject = async (id) => {
  const response = await fetch(`https://color-picker-backend.herokuapp.com/api/v1/projects/${id}`);
  if(!response.ok) {
    return Error('Failed to fetch pallete');
  }
  const palette = await response.json();
  return palette;
};

export const getPalette = async (paletteId) => {
  const response = await fetch(`https://color-picker-backend.herokuapp.com/api/v1/palettes/${paletteId}`);
  if (!response.ok) {
    return Error('Failed to fetch palette');
  }
  const palette = await response.json();
  return palette;
};

export const postProject = async (name) => {
  const response = await fetch(`https://color-picker-backend.herokuapp.com/api/v1/projects`, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify({ name }),
  });
  if (!response.ok) {
    return Error('Failed to post new project');
  }
  const message = await response.json();
  console.log('message', message)
  return message;
};

export const postPalette = async (project_id, palette_name, color_1, color_2, color_3, color_4, color_5) => {
  const response = await fetch('https://color-picker-backend.herokuapp.com/api/v1/palettes', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(
      {
        project_id,
        palette_name,
        color_1,
        color_2,
        color_3,
        color_4,
        color_5,
      },
    ),
  });
  if (!response.ok) {
    return response;
  }
  const message = await response.json();
  return message;
};

export const deleteProject = async (id) => {
  const response = await fetch(`https://color-picker-backend.herokuapp.com/api/v1/projects/${id}`, {
    method: 'DELETE',
  });
  if (!response.ok) {
    return Error('Failed to delete project');
  }
  const message = await response.json();
  return message;
};

export const deletePalette = async (id) => {
  const response = await fetch(`https://color-picker-backend.herokuapp.com/api/v1/palettes/${id}`, {
    method: 'DELETE',
  });
  if (!response.ok) {
    return Error('Failed to delete palette');
  }
  const message = await response.json();
  return message;
};

export const patchProject = async () => {

};

export const patchPalette = async () => {

};
