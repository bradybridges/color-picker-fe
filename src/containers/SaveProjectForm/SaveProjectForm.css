.project-form-container {
  text-align: center;
}

.project-name-input {
  width: 20%;
  background: black;
  color: white;
  height: 30px;
  border-radius: 3px 3px 3px 3px;
  border-color: #2d9fcc;
}

.submit-project-btn{
  margin: 0 auto;
  background: black;
  color: #2d9fcc;
  font-size: 1em;
  border-radius: 3px 3px 3px 3px;
  font-family: 'Permanent Marker';
  border: 1px solid grey;
  width: 20%;
  height: 35px;
  border-color: #2d9fcc;
}
