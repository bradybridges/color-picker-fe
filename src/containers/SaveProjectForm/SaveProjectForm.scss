.project-form-container {
  margin: 10px;
  padding: 10px;
  text-align: center;
}

.project-form {
  display: inline;
}

.project-name-input {
  width: 25%;
  background: black;
  color: white;
  height: 36px;
  border-radius: 3px 3px 3px 3px;
  border-color: #2d9fcc;
}

.submit-project-btn{
  /* margin: 0 auto; */
  background: black;
  border: 1px solid grey;
  border-color: #2d9fcc;
  border-radius: 3px 3px 3px 3px;
  color: #2d9fcc;
  font-family: 'Permanent Marker';
  font-size: 1em;
  height: 35px;
  margin: 7px 5px;
  width: 20%;
  &:hover {
    box-shadow: 3px -3px 6px 0px #2d9fcc;
    cursor: pointer;
  }
}
