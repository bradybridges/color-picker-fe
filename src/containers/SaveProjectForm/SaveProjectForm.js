import React, { Component } from 'react'
import { connect } from 'react-redux'
import { addNewProject } from '../../actions/index'
import './SaveProjectForm.css'

export class SaveProjectForm extends Component {
  constructor() {
    super();
    this.state = {
      name: ''
    }
  }


  handleChange = e => {
    e.preventDefault();
    this.setState({
      [e.target.name]: e.target.value
    })
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    const { name } = this.state;
    const { addNewProject } = this.props
    const response = await this.props.postProject(name)
    const id = await response.id
    console.log('id', id)
    addNewProject(name)
  }


  render() {
    return (
      <div className='project-form-container'>
        <form className='project-form'>
          <input
            name='name'
            value={this.state.name}
            onChange={this.handleChange}
            placeholder='Enter project name'
            className='project-name-input'
          />
        </form>
        <button className='submit-project-btn' onClick={(e) => this.handleSubmit(e)}>Submit New Project</button>
      </div>
    )
  }
}



export const mapState = (state) => ({
  projects: state.projects
})

export const mapDispatch = (dispatch) => ({
  addNewProject: (project) => dispatch(addNewProject(project))
})

export default connect(mapState, mapDispatch)(SaveProjectForm);