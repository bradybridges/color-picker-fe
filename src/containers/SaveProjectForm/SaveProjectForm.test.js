import React from 'react'
import { shallow } from 'enzyme'
import { SaveProjectForm, mapState, mapDispatch } from './SaveProjectForm'
import { addNewProject } from '../../actions'
import { postProject } from '../../apiCalls'


jest.mock('../../apiCalls')

describe('SaveProjectForm', () => {
  describe('SaveProjectForm component', () => {
    let wrapper;

    beforeEach(() => {

      postProject.mockImplementation(() => {
        return Promise.resolve({id: 3})
      })
      const mockProjects = [
        {
          id: 1,
          name: 'a project'
        },
        {
          id: 2,
          name: 'another project'
        }
      ]

      const mockPostProject = jest.fn()
      
      wrapper = shallow(<SaveProjectForm projects={mockProjects} postProject={mockPostProject} />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should update name in state when handleChange is called', () => {
      const mockEvent = { preventDefault() {}, target: {name: 'name', value: 'exciting title'}}

      const expected = 'exciting title'

      wrapper.instance().handleChange(mockEvent)

      expect(wrapper.state('name')).toEqual(expected)
    })

    it('should call the handleSubmit method method when the button is clicked with the correct information', () => {
      wrapper.instance().handleSubmit = jest.fn()
      wrapper.instance().forceUpdate()

      wrapper.find('button').simulate('click')

      expect(wrapper.instance().handleSubmit).toHaveBeenCalled()
    })

    it.skip('should call handleSubmit with the correct information', () => {
      const mockEvent = { preventDefault: jest.fn() }

      wrapper.instance().setState({'name': 'rainbows and unicorns'})

      wrapper.instance().addNewProject = jest.fn()

      wrapper.instance().handleSubmit = jest.fn()

      wrapper.instance().handleSubmit(mockEvent)

      expect(wrapper.instance().handleSubmit).toHaveBeenCalled


    })
  })
})


// it('call handleDeleteProject with the correct information', () => {
//   const mockEvent = { preventDefault: jest.fn() }
  
//   wrapper.instance().updateProjects = jest.fn()
//   wrapper.instance().updatePalettes = jest.fn()

//   wrapper.instance().handleDeleteProject(mockEvent)

//   expect(deleteProject).toHaveBeenCalledWith(1);

//   expect(wrapper.instance().updateProjects).toHaveBeenCalledWith(1, [{"id": 1, "name": "a project"}, {"id": 2, "name": "another project"}])

//   expect(wrapper.instance().updatePalettes).toHaveBeenCalledWith(1, [
//     {
//       id: 1,
//       project_id: 1,
//       name: 'a palette',
//       color_1: '#123456',
//       color_2: '#123456',
//       color_3: '#123456',
//       color_4: '#123456',
//       color_5: '#123456',
//     }, 
//     {
//       id: 2,
//       project_id: 1,
//       name: 'another palette',
//       color_1: '#AABBCC',
//       color_2: '#AABBCC',
//       color_3: '#AABBCC',
//       color_4: '#AABBCC',
//       color_5: '#AABBCC',
//     }
//   ])
  


// })



// it('should call the submitForm method when the button is clicked', () => {
//   wrapper.instance().submitForm = jest.fn()
//   wrapper.instance().forceUpdate()

//   wrapper.find('button').simulate('click')

//   expect(wrapper.instance().submitForm).toHaveBeenCalled()
// })

// it.skip('should call the submitForm method with the properties in state when the button is clicked', () => {
//   const mockTitle = 'a title'
//   const mockPlot = 'a plot'

//   wrapper.instance().submitForm = jest.fn()

//   wrapper.instance().setState({
//     title: mockTitle,
//     plot: mockPlot
//   })

//   wrapper.find('button').simulate('click')

//   expect(wrapper.instance().submitForm).toHaveBeenCalledWith(mockTitle, mockPlot)
// })
// })