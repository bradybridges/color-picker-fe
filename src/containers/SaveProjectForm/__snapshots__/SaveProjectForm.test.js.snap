// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveProjectForm SaveProjectForm component should match the snapshot 1`] = `
<div
  className="project-form-container"
>
  <form
    className="project-form"
  >
    <input
      className="project-name-input"
      name="name"
      onChange={[Function]}
      placeholder="Enter project name"
      value=""
    />
  </form>
  <button
    className="submit-project-btn"
    onClick={[Function]}
    type="button"
  >
    Submit New Project
  </button>
</div>
`;
